src/Bicep.Core/Analyzers/Linter/Rules/NoUnnecessaryDependsOnRule.cs

```diff
@@ -79,37 +81,67 @@ public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax sy
+                    }
+
+                    if (!inferredDependencies.Any(d => d.Resource == referencedResource))
+                    {
+                        // There are no inferred dependencies - the dependsOn entry is valid
+                        continue;
+                    }
+
+                    CodeReplacement? codeReplacement = null;
+                    if (declaredDependencies.Items.Count() == 1)
                     {
-                        // If this resource has no implicit dependencies, than all explicit dependsOn entries must be valid, so don't bother checking
-                        if (inferredDependenciesMap.Value.TryGetValue(thisResource, out ImmutableHashSet<ResourceDependency>? inferredDependencies))
+                        // we only have one entry - remove the whole dependsOn property
+                        if (SyntaxModifier.TryRemoveProperty(body, dependsOnProperty) is {} newObject)
                         {
-                            foreach (ArrayItemSyntax declaredDependency in declaredDependencies.Items)
-                            {
-                                // Is this a simple reference to a resource collection?
-                                if (model.GetSymbolInfo(declaredDependency.Value) is ResourceSymbol referencedResource)
-                                {
-                                    if (referencedResource.IsCollection)
-                                    {
-                                        // Ignore dependsOn entries pointing to a resource collection - dependency analyis would
-                                        // be complex and user probably knows what they're doing.
-                                        continue;
-                                    }
-
-                                    if (inferredDependencies.Any(d => d.Resource == referencedResource))
-                                    {
-                                        this.diagnostics.Add(
-                                            parent.CreateDiagnosticForSpan(
-                                                diagnosticLevel,
-                                                declaredDependency.Span,
-                                                referencedResource.Name));
-                                    }
-                                }
-                            }
+                            codeReplacement = new CodeReplacement(body.Span, newObject.ToTextPreserveFormatting());
                         }
                     }
+                    else
+                    {
+                        // we have multiple entries - just remove this one
+                        if (SyntaxModifier.TryRemoveItem(declaredDependencies, declaredDependency) is {} newArray)
+                        {
+                            codeReplacement = new CodeReplacement(declaredDependencies.Span, newArray.ToTextPreserveFormatting());
+                        }
```

Comment:
Is it worth adding a branch to emit a code fix if `dependsOn` is empty? If a user had two unnecessary dependencies and applied both code fixes, they could end up with `dependsOn: []` (assuming the second fix could be applied before a new fix removing the single member `dependsOn` property were generated).

Reply:
All of our code fixes are generated based on absolute positioning (a span to be replaced + the replacement text), so they cannot be safely applied on top of each other, and we would instead need to wait for an updated semantic model.